modules/rostests/apitests/ntdll/probelib.c

```diff
@@ -31,21 +31,31 @@ QuerySetProcessValidator(
                     break;
                 }
 
+                /*
+                 * NOTE: Buffer validation checks are relaxed on Windows Server 2003 SP2,
+                 * that is, the system only checks if the pointer is properly aligned in
+                 * NtQueryInformationProcess function and friends but it doesn't check if the
+                 * buffer is valid (that is up for each information class to care about that, whatever
+                 * that is). On ReactOS however we're way aggressive regarding this.
```

Comment:
Isn't what you were implementing in e.g. DefaultQueryInfoBufferCheck() with this ICIF_QUERY_SIZE_VARIABLE ?
If so you could always move that check after the probing done in DefaultQueryInfoBufferCheck() ?

Reply:
Dunno what exactly you want with this but your suggestion actually breaks stuff even more. `ICIF_QUERY_SIZE_VARIABLE` means the size length of an information class is arbitrary, whereas the lengths whose sizes are fixed mustn't have this flag. As this flag and similar were badly misused in the tables, that reason alone is why the tests failed previously.

Moving this flag check after the buffer probing is done, as you suggested, the buffer is always probed first and size length is checked after and that makes the tests fail again. Btw, I removed that comment in code, so forget about it.

Reply:
ok, thanks for the explanation.